client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/OperationChannel.java

```diff
@@ -81,29 +81,33 @@ CompletionStage<Void> attemptConnect() {
       }
       channel = null;
       ChannelFuture channelFuture = newChannelInvoker.createChannel();
-      // It is possible for channelActive calls and this to be done in non deterministic ordering
-      channelFuture.addListener(f -> {
-         if (f.isSuccess()) {
-            Channel c = channelFuture.channel();
-            assert c.eventLoop().inEventLoop();
-            c.attr(OPERATION_CHANNEL_ATTRIBUTE_KEY).set(this);
-            channel = c;
-            headerDecoder = c.pipeline().get(HeaderDecoder.class);
-            codec = headerDecoder.getConfiguration().version().getCodec();
-            connectionFailureListener.accept(this, null);
-            // Now we can let auth or operations to continue
-            channel.pipeline().fireUserEventTriggered(ActivationHandler.ACTIVATION_EVENT);
-            log.tracef("OperationChannel %s connect complete to %s", this, c);
-         } else {
-            Throwable cause = f.cause();
-            log.tracef("Connection attempt to %s encountered exception for %s", address, cause);
-            // Allow another attempt later
-            handleError(connectFuture, cause);
-         }
-      });
       // This check here is in case if there is an issue with the event loop executor where it doesn't actually run
       // the listener above, which can happen if the event loop group is shutdown
       Throwable immediateError = channelFuture.cause();
+      if (immediateError == null) {
```

Comment:
I guess the PR needs a rebase to pull #13742 changes in?

Reply:
yup, I broke it out to make it easier to integrate :)